### DISCUSSION TITLE
[Feature] Showing and filtering existing tags

### DISCUSSION THREAD

**Discussion by faroinsua** [immich-app/immich#3950] [Feature Request]
### The feature

I think filtering by tags contained in the exif-data is an important feature when managing huge foto collections. 
https://github.com/xemle/home-gallery
is simple but has this feature. Is there a chance to be included some day?

It would avoid buying expensive NAS Software. 

### Platform

- [ ] Server
- [X] Web
- [X] Mobile

**Comment by charles-997** [comment]
Similar to: https://github.com/immich-app/immich/discussions/2168

**Comment by imler** [comment]
Bear in mind when implementing that tags / keywords are stored in different fields in metadata: IPTC - "Keywords", XMP - "Subject", and EXIF - "XP Keywords" to name these at least.

All of them contain a list of keywords describing the content of the photo, but the list of IPTC and XMP lists are comma-separated, however EXIF list is separated by semicolons.

All of those keywords should be imported during photo upload and then be searchable and visible in the details pane of each photo. Clicking on a tag should transition to search results or any other page that would show all photos with a specific keyword.

Another section with keywords should be also present on the "Explore" page.

**Reply by GD-Dal** [reply]
Thanks. This is sorely needed.
Is it possible to search ANY metadata as per now?
I have tried, and can't make a single hit of any of the metadata inside the file. Nor in IPTC or in EXIF

**Comment by psyciknz** [comment]
Include any metadata exposed in xmp sidecar files

**Comment by bo0tzz** [comment]
Will be closed by #11980.
